makefiles/master/rules.mk

```diff
@@ -18,18 +21,43 @@ THEOS_USE_PARALLEL_BUILDING := $(_THEOS_IS_MAKE_GT_4_0)
 endif
 export THEOS_USE_PARALLEL_BUILDING
 
-ifeq ($(MAKELEVEL)$(call __theos_bool,$(THEOS_USE_PARALLEL_BUILDING)),0$(_THEOS_TRUE))
+# This is effectively THEOS_USE_PARALLEL_BUILDING canonicalized
+ifeq ($(_THEOS_INTERNAL_USE_PARALLEL_BUILDING),)
+_THEOS_INTERNAL_USE_PARALLEL_BUILDING := $(call __theos_bool,$(THEOS_USE_PARALLEL_BUILDING))
+endif
+export _THEOS_INTERNAL_USE_PARALLEL_BUILDING
+
+# certain conditions need to execute, semantically, when we're
+# running the first `all`. This is usually when MAKELEVEL == 0
+# but is in fact MAKELEVEL == 1 if we're running `troubleshoot`
+_THEOS_TOP_ALL_MAKELEVEL := $(if $(THEOS_IS_TROUBLESHOOTING),1,0)
+ifeq ($(MAKELEVEL),$(_THEOS_TOP_ALL_MAKELEVEL))
+_THEOS_IS_TOP_ALL := $(_THEOS_TRUE)
+endif
```

Comment:
Not sure if we should make this assumption in this way, since we can be called by another non-Theos makefile too. Maybe we could look for something we export later on?

Reply:
The true semantic intent here is to recreate the auxiliary/markers directories once at the start of a build — I wonder if there's a more Make-esque way to do this using some sort of rule/intermediate. IMO this is good enough for now but I can follow-up with a better approach when I have the chance.